### DISCUSSION TITLE
When to use @cucumber/cucumber and @wdio/cucumber-framework?

### DISCUSSION THREAD

**Discussion by testerdv** [webdriverio/webdriverio#10051] [Q&A]
From the wdio docs v8 from [here](https://webdriver.io/docs/frameworks#cucumber-options) it says:

> **Import Step Definition Helper**
> 
> In order to use step definition helper like Given, When or Then or hooks, you are suppose to import then from @cucumber/cucumber, e.g. like this:
> 
> import { Given, When, Then } from '@cucumber/cucumber'
> 
> Now, if you use Cucumber already for other types of tests unrelated to WebdriverIO for which you use a specific version you need to import these helpers in your e2e tests from the WebdriverIO Cucumber package, e.g.:
> 
> import { Given, When, Then } from '@wdio/cucumber-framework'
> 
> This ensures that you use the right helpers within the WebdriverIO framework and allows you to use an independant Cucumber version for other types of testing.

Are both code statements written in opposite way? or What it wants to convey I could not understand it. What is its meaning? when to use

`import { Given, When, Then } from '@cucumber/cucumber'`

and when to use

`import { Given, When, Then } from '@wdio/cucumber-framework'`?

**Comment by christian-bromann** [comment]
I would always use `import { Given, When, Then } from '@wdio/cucumber-framework'` because it guarantees that you use the Cucumber functions that WebdriverIO uses for integrating with Cucumber. If you e.g. use a Cucumber version for other types of testing that is 1 version behind of the one used by WebdriverIO, you can experience problems.
